eth-node/bridge/geth/wakuv2.go

```diff
@@ -338,14 +306,75 @@ func (w *gethWakuV2Wrapper) ConfirmMessageDelivered(hashes []common.Hash) {
 	w.waku.ConfirmMessageDelivered(hashes)
 }
 
-func (w *gethWakuV2Wrapper) SetStorePeerID(peerID peer.ID) {
-	w.waku.SetStorePeerID(peerID)
-}
-
 func (w *gethWakuV2Wrapper) PeerID() peer.ID {
 	return w.waku.PeerID()
 }
 
-func (w *gethWakuV2Wrapper) PingPeer(ctx context.Context, peerID peer.ID) (time.Duration, error) {
-	return w.waku.PingPeer(ctx, peerID)
+func (w *gethWakuV2Wrapper) GetActiveStorenode() peer.ID {
+	return w.waku.StorenodeCycle.GetActiveStorenode()
+}
+
+func (w *gethWakuV2Wrapper) OnStorenodeAvailableOneShot() <-chan struct{} {
+	return w.waku.StorenodeCycle.StorenodeAvailableOneshotEmitter.Subscribe()
+}
+
+func (w *gethWakuV2Wrapper) OnStorenodeChanged() <-chan peer.ID {
+	return w.waku.StorenodeCycle.StorenodeChangedEmitter.Subscribe()
+}
+
+func (w *gethWakuV2Wrapper) OnStorenodeNotWorking() <-chan struct{} {
+	return w.waku.StorenodeCycle.StorenodeNotWorkingEmitter.Subscribe()
+}
+
+func (w *gethWakuV2Wrapper) OnStorenodeAvailable() <-chan peer.ID {
+	return w.waku.StorenodeCycle.StorenodeAvailableEmitter.Subscribe()
+}
+
+func (w *gethWakuV2Wrapper) WaitForAvailableStoreNode(timeout time.Duration) bool {
+	return w.waku.StorenodeCycle.WaitForAvailableStoreNode(context.TODO(), timeout)
+}
```

Comment:
`context` should probably passed as argument to `WaitForAvailableStoreNode`
```suggestion
func (w *gethWakuV2Wrapper) WaitForAvailableStoreNode(ctx context.Context, timeout time.Duration) bool {
	return w.waku.StorenodeCycle.WaitForAvailableStoreNode(ctx, timeout)
}
```